app/models/user_subscription.rb

```diff
@@ -4,12 +4,19 @@
 class UserSubscription < ApplicationRecord
   ALLOWED_TYPES = %w[Article].freeze
 
-  counter_culture :subscriber, column_name: "subscribed_to_user_subscriptions_count"
-  counter_culture :user_subscription_sourceable, column_name: "user_subscriptions_count"
-
   belongs_to :author, class_name: "User", inverse_of: :source_authored_user_subscriptions
   belongs_to :subscriber, class_name: "User", inverse_of: :subscribed_to_user_subscriptions
+  counter_culture :subscriber,
+                  column_name: nil,
+                  column_names: {
+                    true => "subscribed_to_user_subscriptions_count"
+                  }
   belongs_to :user_subscription_sourceable, polymorphic: true, optional: true
+  counter_culture :user_subscription_sourceable,
+                  column_name: nil,
+                  column_names: {
+                    true => "user_subscriptions_count"
+                  }
```

Comment:
This counter cache issue ends up taking way more time than I thought because I went down the route of not using `counter_culture`, but I ran into the issue of not being able to track destroy the item, then @rhymes point out to me a neat way avoid the callback without removing `counter_culture`. 

I am almost there. I just can't figure out why it's not updating the counter properly when a subscription_sourceable is destroy.

Reply:
Neat work around!

Reply:
After a bit of messing around, I found out that `UserSubscription.counter_culture_fix_counts` won't be able to properly decrement a polymorphic's count to 0. It can never go below 1 because of [counter_culture's reconciliation strategy](https://github.com/magnusvk/counter_culture/blob/master/lib/counter_culture/reconciler.rb#L48). It works by looking for a distinct `*_type` attributes and it won't be able to when there's no record to work with. This still remains the same if I were to properly use `column_name: :user_subscriptions_count` because what decrement the counter is the AR callback. 

If this is an acceptable side-effect I can adjust the test to make this pass. The other idea I have that would require more work would be to move the whole `create` process into a background job. I'm happy to dive further and am all ears for any suggestion.